File: Pgms/Semaforo03VB/TestSemaforo.vb
'
' TestSemaforo.vb
' Created by SharpDevelop.
' User: LAOS
' Date: 9/14/2004
' Time: 10:15 PM
' 
'

Imports System

Namespace CursoApliNet

	Public Class Principal

		Shared Sub Main(ByVal args As String())

			' Comprueba dos argumentos
			If args.Length < 2 Then
				Console.WriteLine
				Console.WriteLine("Forma de uso: test segsSemaforo segsFuncionando")
				Return
			End If

			' Obtiene datos para el programa
			Dim segsSemaforo As Integer = Int32.Parse(args(0))
			Dim segsFuncionando As Integer = Int32.Parse(args(1))

			' Creacion de semaforo e inicializacion
			Dim smf As SemaforoN = New SemaforoN 
			smf.SetSegundos(segsSemaforo)
			smf.SetEstados(5)

			' Suscripcion al evento cambio estado
			AddHandler smf.CambioEstado, AddressOf DespliegaDia

			' Pone en funcionamiento el semaforo
			smf.Run(segsFuncionando)

		End Sub

		Public Shared Sub DespliegaDia(ByVal semaforo As Object, ByVal estadoInfo As EstadoEventArgs)

			' Traduce estado a dia
			Dim dia As String = EstadoADia(estadoInfo.Estado)
			
			' Despliega dia
			Console.WriteLine(dia)

		End Sub

		Shared Function EstadoADia(ByVal estado As Integer) As String

			Dim dia As String = ""
			Select estado
			Case 0
				dia = "Lunes"
			Case 1
				dia = "Martes"
			Case 2
				dia = "Miercoles"
			Case 3
				dia = "Jueves"
			Case Else
				dia = "Viernes"
			End Select 

			Return dia

		End Function

	End Class 

End Namespace

File: Pgms/Semaforo03CS/TestSemaforo.cs
/*
 * TestSemaforo.cs
 * Created by SharpDevelop.
 * User: LAOS
 * Date: 9/14/2004
 * Time: 10:15 PM
 * 
 */

using System;

namespace CursoApliNet
{
	public class Principal
	{
		static void Main(string[] args)
		{
			// Comprueba dos argumentos
			if(args.Length < 2)
			{
				Console.WriteLine();
				Console.WriteLine("Forma de uso: test segsSemaforo segsFuncionando");
				return;
			}
		
			// Obtiene datos para el programa
			int segsSemaforo = Int32.Parse(args[0]);
			int segsFuncionando = Int32.Parse(args[1]);
			
			// Creacion de semaforo e inicializacion
			SemaforoN smf = new SemaforoN();
			smf.SetSegundos(segsSemaforo);
			smf.SetEstados(5);
		
			// Suscripcion al evento cambio estado
			smf.CambioEstado += new CambioEstadoHandler(DespliegaDia);
			
			// Pone en funcionamiento el semaforo
			smf.Run(segsFuncionando);
		}
		
       	static public void DespliegaDia(object semaforo, EstadoEventArgs estadoInfo)
       	{
			// Traduce estado a dia
			string dia = EstadoADia(estadoInfo.Estado);
       		
			// Despliega dia
            Console.WriteLine(dia);  
       	}     
		
		static string EstadoADia(int estado)
		{
			string dia = "";
			switch(estado)
			{
				case 0:
					dia = "Lunes";
					break;
				case 1:
					dia = "Martes";
					break;
				case 2:
					dia = "Miercoles";
					break;
				case 3:
					dia = "Jueves";
					break;
				default:
					dia = "Viernes";
					break;
			}
			
			return dia;
		}
	}
}
